src/bitcoind/poller.rs

```diff
@@ -1394,7 +1417,10 @@ fn handle_confirmed_deposit(
         &cancel_tx,
         emer_tx.as_ref(),
         unemer_tx.as_ref(),
-    )?;
+    )
+    .unwrap_or_else(|e| {
```

Comment:
I don't know. Here are two reasons to panic on the call site:
1. You know what you panic on. Forwarding the errors and `unwrap()`ing at the root of the call chain makes it much harder to reason about what you are panic'ing on.
2.  the backtraces really aren't perfect (they currently only identify the line where you panic)

I guess 1. could be mitigated using enum variants (like, panic only on DbError).

Another reason to not panic in place is for flexibility, for instance to test the errors in unit tests. Instead of the more clumsy and less flexible "should panic" testing strategy.

Reply:
Cool if we want to move to another way of handling errors, we could discuss it (and I would bring on the table `anyhow` and `thiserror` again, as I did one year ago, you're warned 😄), but I don't think this PR is the right place to do so. I would keep the `?` for consistency

Reply:
Very much agreed. I'll remove that debugging patch.

Reply:
> and I would bring on the table anyhow and thiserror again

Not sure how this is related btw

Reply:
No i want to panic everywhere because i know what i panic on and because of poor backtraces